frontend/src/context/Metrics/metricsSlice.tsx

```diff
@@ -266,8 +281,23 @@ export const metricsSlice = createSlice({
               step: '',
             }))
 
+      const getPipelinesWarningMessage = (pipelines: IPipelineConfig[]) =>
```

Comment:
so complex, could u split it into multiple functions??

Reply:
yes, I've extracted a function to create warning message for deployment and leadTime